tests/LSP6KeyManager/tests/PermissionsChangeAddURD.test.ts

```diff
@@ -0,0 +1,1353 @@
+            .execute(payload);
+
+          const result = await context.universalProfile["getData(bytes32)"](
+            payloadParam.dataKey
+          );
+          expect(ethers.utils.getAddress(result)).to.equal(
+            payloadParam.dataValue
+          );
+        });
+
+        it("should be allowed to CHANGE a UniversalReceiverDelegate key", async () => {
+          const payloadParam = {
+            dataKey: universalReceiverDelegateKey1,
+            dataValue: universalReceiverDelegateB.address,
+          };
+
+          let payload = context.universalProfile.interface.encodeFunctionData(
+            "setData(bytes32,bytes)",
+            [payloadParam.dataKey, payloadParam.dataValue]
+          );
+
+          await context.keyManager
+            .connect(canAddAndChangeUniversalReceiverDelegate)
+            .execute(payload);
+
+          const result = await context.universalProfile["getData(bytes32)"](
+            payloadParam.dataKey
+          );
+          expect(ethers.utils.getAddress(result)).to.equal(
+            payloadParam.dataValue
+          );
+        });
+
+        it("should be allowed to CHANGE (remove) a UniversalReceiverDelegate key set", async () => {
+          const payloadParam = {
+            dataKey: universalReceiverDelegateKey1,
+            dataValue: "0x",
+          };
+
+          let payload = context.universalProfile.interface.encodeFunctionData(
+            "setData(bytes32,bytes)",
+            [payloadParam.dataKey, payloadParam.dataValue]
+          );
+
+          await context.keyManager
+            .connect(canAddAndChangeUniversalReceiverDelegate)
+            .execute(payload);
+
+          const result = await context.universalProfile["getData(bytes32)"](
+            payloadParam.dataKey
+          );
+          expect(result).to.equal(payloadParam.dataValue);
+        });
+      });
+
+      describe("when caller is an address with ADDUniversalReceiverDelegate permission", () => {
+        it("should be allowed to ADD a UniversalReceiverDelegate key", async () => {
+          const payloadParam = {
+            dataKey: ERC725YKeys.LSP1.LSP1UniversalReceiverDelegate,
+            dataValue: universalReceiverDelegateA.address,
+          };
+
+          let payload = context.universalProfile.interface.encodeFunctionData(
+            "setData(bytes32,bytes)",
+            [payloadParam.dataKey, payloadParam.dataValue]
+          );
+
+          await context.keyManager
+            .connect(canOnlyAddUniversalReceiverDelegate)
+            .execute(payload);
+
+          const result = await context.universalProfile["getData(bytes32)"](
+            payloadParam.dataKey
+          );
+          expect(ethers.utils.getAddress(result)).to.equal(
+            payloadParam.dataValue
+          );
+        });
+
+        it("shouldn't be allowed to CHANGE the UniversalReceiverDelegate key set even if it's setting existing data", async () => {
```

Comment:
```suggestion
        it("should NOT be allowed to CHANGE the UniversalReceiverDelegate key set even if it's setting existing data", async () => {
```

>  even if it's setting existing data

what does that mean?